static/panes/compiler.ts

```diff
@@ -1380,7 +1383,11 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
         // After a short delay, give the user some indication that we're working on their
         // compilation.
         const progress = setTimeout(() => {
-            this.setAssembly({asm: this.fakeAsm('<Compiling...>')}, 0);
+            if (!_.isEqual(this.assembly, COMPILING_PLACEHOLDER)) {
```

Comment:
Can you refactor this into a function, as the code below is the same.
Perhaps even including the `setTimeout`, as thats the same too and then we don't need the comment

Reply:
Good idea